pkg/database/db_manager.go

```diff
@@ -266,10 +266,15 @@ func (m *DBManager) Has(name string) bool {
 }
 
 func (m *DBManager) HasIndex(idx int) bool {
-	m.dbMutex.RLock()
-	defer m.dbMutex.RUnlock()
+	db, exists := m.getDB(idx)
+	if !exists {
+		return false
+	}
+
+	db.mtx.Lock()
+	defer db.mtx.Unlock()
 
-	return idx >= 0 && idx < len(m.databases)
+	return !db.deleted
```

Comment:
Shouldn't be here also check for idx?

Reply:
Alredy done inside function `db, exists := m.getDB(idx)`. It returns `false` if `idx` is not inside the range